src/main.rs

```diff
@@ -129,14 +134,25 @@ fn run_machete() -> anyhow::Result<bool> {
     let args = parse_args()?;
 
     for path in args.paths {
-        let manifest_path_entries = collect_paths(&path, args.skip_target_dir);
+        let toml_file_iter = build_toml_file_iterator(&path, args.skip_target_dir);
 
         // Run analysis in parallel. This will spawn new rayon tasks when dependencies are effectively
         // used by any Rust crate.
-        let results = manifest_path_entries
-            .par_iter()
-            .filter_map(
-                |manifest_path| match find_unused(manifest_path, args.use_cargo_metadata) {
+        let results = toml_file_iter
+            .filter_map(|entry| match entry {
+                Ok(entry) => Some(entry.into_path()),
+                Err(error) => {
+                    eprintln!("error when walking over subdirectories: {}", error);
+                    None
+                }
+            })
+            // NOTE(mickvangelderen): Instead of building a parallel iterator through rayon, we
+            // could also use the parallel walker exposed by ignore. We will have to implement a
+            // visitor type to collect the analysis results though. I opted not to do this in the
+            // initial PR to limit the amount of changes.
+            .par_bridge()
+            .filter_map(|manifest_path| {
```

Comment:
We're not walking the file system in parallel while we could. We only process the files in parallel. Not sure if it matters.